pkgs/tools/package-management/lix/common.nix

```diff
@@ -0,0 +1,237 @@
+
+, confDir
+, stateDir
+, storeDir
+}: let
+  lix-doc = callPackage ./doc {
+    inherit src;
+    version = "${version}${suffix}";
+    cargoHash = docCargoHash;
+  };
+  self = stdenv.mkDerivation {
+    pname = "lix";
+
+    version = "${version}${suffix}";
+    VERSION_SUFFIX = suffix;
+
+    inherit src patches;
+
+    outputs =
+      [ "out" "dev" ]
+      ++ lib.optionals enableDocumentation [ "man" "doc" ];
+
+    hardeningEnable = lib.optionals (!stdenv.isDarwin) [ "pie" ];
+
+    hardeningDisable = lib.optional stdenv.hostPlatform.isMusl "fortify";
+
+    nativeBuildInputs = [
+      pkg-config
+      bison
+      flex
+      jq
+      meson
+      ninja
+      cmake
+      python3
+      doxygen
+
+      # Tests
+      git
+      mercurial
+      jq
+      lsof
+    ] ++ lib.optionals (enableDocumentation) [
+      (lib.getBin lowdown)
+      mdbook
+      mdbook-linkcheck
+    ] ++ lib.optionals stdenv.isLinux [
+      util-linuxMinimal
+    ];
+
+    buildInputs = [
+      boost
+      brotli
+      bzip2
+      curl
+      editline
+      libsodium
+      openssl
+      sqlite
+      xz
+      gtest
+      libarchive
+      lowdown
+      rapidcheck
+      toml11
+      lix-doc
+    ] ++ lib.optionals stdenv.isDarwin [
+      Security
+    ] ++ lib.optionals (stdenv.isx86_64) [
+      libcpuid
+    ] ++ lib.optionals withLibseccomp [
+      libseccomp
+    ] ++ lib.optionals withAWS [
+      aws-sdk-cpp
+    ];
+
+    propagatedBuildInputs = [
+      boehmgc
+      nlohmann_json
+    ];
+
+    postPatch = ''
+      patchShebangs --build tests
+    '';
+
+    preConfigure =
+      # Copy libboost_context so we don't get all of Boost in our closure.
+      # https://github.com/NixOS/nixpkgs/issues/45462
+      lib.optionalString (!enableStatic) ''
+        mkdir -p $out/lib
+        cp -pd ${boost}/lib/{libboost_context*,libboost_thread*,libboost_system*} $out/lib
+        rm -f $out/lib/*.a
+        ${lib.optionalString stdenv.isLinux ''
+          chmod u+w $out/lib/*.so.*
+          patchelf --set-rpath $out/lib:${stdenv.cc.cc.lib}/lib $out/lib/libboost_thread.so.*
+        ''}
+      '';
+
+    mesonFlags = [
+      (lib.mesonEnable "gc" true)
+      (lib.mesonBool "enable-docs" enableDocumentation)
+      (lib.mesonBool "enable-embedded-sandbox-shell" (stdenv.isLinux && stdenv.hostPlatform.isStatic))
+      (lib.mesonEnable "seccomp-sandboxing" withLibseccomp)
+
+      (lib.mesonOption "store-dir" storeDir)
+      (lib.mesonOption "state-dir" stateDir)
+      (lib.mesonOption "sysconfdir" confDir)
+    ] ++ lib.optionals stdenv.isLinux [
+      (lib.mesonOption "sandbox-shell" "${busybox-sandbox-shell}/bin/busybox")
+    ];
+
+    # Needed for Meson to find Boost.
+    # https://github.com/NixOS/nixpkgs/issues/86131.
+    env = {
+      BOOST_INCLUDEDIR = "${lib.getDev boost}/include";
+      BOOST_LIBRARYDIR = "${lib.getLib boost}/lib";
+    };
+
+    installFlags = [ "sysconfdir=$(out)/etc" ];
+
+    doInstallCheck = true;
```

Comment:
Nixpkgs doesn't provide a `mesonInstallCheckPhase` or `ninjaInstallCheckPhase`, so you'll need to override `installCheckPhase` yourself or the functional tests won't be run

Reply:
Yep, this is done now.